arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorHolder.java

```diff
@@ -146,4 +150,17 @@ public PositionVectorHolder(FieldVector vector, Type type, NullabilityHolder nul
     }
   }
 
+  public static class DeletedVectorHolder extends VectorHolder {
```

Comment:
nit: a little weird to me to have an extended class nested inside

Reply:
I guess it's fine, since we've got nested classes `PositionVectorHolder` and `ConstantVectorHolder` extend `VectorHolder`.

Reply:
agree. it is existing pattern and we don't need to change. regardless, I just thought it a little weird